Branch-SDK/Branch-SDK/BranchNotifications.h

```diff
@@ -0,0 +1,31 @@
+//
+//  BranchNotifications.h
+//  Branch-TestBed
+//
+//  Created by edward on 6/30/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+@class Branch;
+
+@protocol BranchDelegate <NSObject>
```

Comment:
1.  I agree that untyped data is not a good idea.  I think changing these notifications to a BUO is a great idea.  Have untyped data is hard for the SDK user, and we are trying to move everything to a BUO anyway.

2. Thanks for the feedback on semantic naming of the notifications / delegate methods.

Maybe these names are better:

`branch:willOpenURL:`  (When URL is non-null).
`branchWillCheckForDeferredDeepLink:` (When URL is null).

`branch:didOpenBranchUniversalObject:withLinkProperties:`
`branch:failedToOpenURL:withError:`

what other events and notifications would be helpful?  How should they be named?